configs/components/yaml-cpp.rb

```diff
@@ -26,6 +26,13 @@
     if platform.is_cross_compiled?
       pkg.environment 'CXX', 'clang++ -target arm64-apple-macos11' if platform.name =~ /osx-11/
       pkg.environment 'CXX', 'clang++ -target arm64-apple-macos12' if platform.name =~ /osx-12/
+    else
+      pkg.environment 'CXX', 'clang++'
+    end
+
+    # If the architecture of the mac OS is arm64, the cmake path should be set to homebrew bin
+    if platform.architecture == "arm64"
+      cmake = "/opt/homebrew/bin/cmake"
```

Comment:
This is only true for macOS 13 ARM (or greater), but not when cross compiling macOS 11 & 12 for ARM. My suggestion:

```ruby
    if platform.is_cross_compiled?
      pkg.environment 'CXX', 'clang++ -target arm64-apple-macos11' if platform.name =~ /osx-11/
      pkg.environment 'CXX', 'clang++ -target arm64-apple-macos12' if platform.name =~ /osx-12/
    else
      pkg.environment 'CXX', 'clang++'
      cmake = "/opt/homebrew/bin/cmake" if platform.architecture == "arm64"
    end
```